tests/test_app.py

```diff
@@ -1003,3 +1005,368 @@ async def test_importer_mappings():
             assert resp.status == 400
             text = await resp.text()
             assert f"must provide file_list field. Your provided qs: {unquote(qsd)}" in text
+
+
+async def test_bulk_specification_success():
+    # In other tests a username is passed to AppClient but AppClient completely ignores it...
+    async with AppClient(config) as cli:
+        with FileUtil() as fu:
+            fu.make_dir("testuser/somefolder")  # testuser is hardcoded in the auth mock
+            base = Path(fu.base_dir) / "testuser"
+            tsv = "genomes.tsv"
+            with open(base / tsv, "w") as f:
+                f.writelines([
+                    "Data type: genomes; Columns: 3; Version: 1\n",
+                    "spec1\tspec2\t   spec3   \n",
+                    "Spec 1\t Spec 2\t Spec 3\n",
+                    "val1 \t   ꔆ   \t    7\n",
+                    "val3\tval4\t1\n",
+                ])
+            csv = "somefolder/breakfastcereals.csv"
+            with open(base / csv, "w") as f:
+                f.writelines([
+                    "Data type: breakfastcereals; Columns: 3; Version: 1\n",
+                    "s1,s2,s3\n",
+                    "S 1,S 2,S 3\n",
+                    "froot loops ,   puffin   ,   gross\n",
+                    "grape nuts , dietary fiber, also gross\n",
+                ])
+            excel = "importspec.xlsx"
+            with pandas.ExcelWriter(base / excel) as exw:
+                df = pandas.DataFrame([
+                    ["Data type: fruit_bats; Columns: 2; Version: 1"],
+                    ["bat_name", "wing_count"],
+                    ["Name of Bat", "Number of wings"],
+                    ["George", 42],
+                    ["Fred", 1.5]
+                ])
+                df.to_excel(exw, sheet_name="bats", header=False, index=False)
+                df = pandas.DataFrame([
+                    ["Data type: tree_sloths; Columns: 2; Version: 1"],
+                    ["entity_id", "preferred_food"],
+                    ["Entity ID", "Preferred Food"],
+                    ["That which ends all", "ꔆ"],
```

Comment:
What the ꔆ

Reply:
Just testing unicode works throughout the stack